### PR TITLE
Fix wrong added filename

### PR DESCRIPTION
**Pull request by camelmasa**
I think It looks `files.NewSerialFile`'s first argument is wrong.

Note: The `coreunix.AddR` will be removed after implemented Core API (Ref: https://github.com/ipfs/go-ipfs/pull/4562).

### CODE REVIEW

**Comment by Kubuxu** [comment]
I am wondering why no tests have detected it.

**Comment by camelmasa** [comment]
@Kubuxu Thanks for checking.

I just changed test data directory path from `test_data` to `test/data`.

Case of original code,  Returned value was changed after I changed just the test data directory path.
Original test case was couldn't detect error when the directory was nested.

File: core/coreunix/add.go [hunk 1]
```diff
@@ -336,7 +337,10 @@ func AddR(n *core.IpfsNode, root string) (key string, err error) {
 		return "", err
 	}
 
-	f, err := files.NewSerialFile(root, root, false, stat)
+	segs := strings.Split(root, "/")
+	name := segs[len(segs)-1]
```
Comment (Kubuxu) [inline core/coreunix/add.go]: use `name := filepath.Base(root)` instead of string splitting.
Filepath is aware of system specific separators.

Comment (magik6k) [reply core/coreunix/add.go]: This probably broke `go test` on windows - https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fgo-ipfs/detail/PR-4919/4/pipeline

**Comment by whyrusleeping** [comment]
Since the author has been gone for ~1mo, can someone else take over this changeset? cc @djdv maybe?

**Comment by djdv** [comment]
Amended in: https://github.com/ipfs/go-ipfs/pull/5167

**Pull request closed without merging**